File: Tutor Demos/Week 7/Java/PartnershipOffer.java
public class PartnershipOffer implements Request {

    private String name;

    public PartnershipOffer(String name) {
        this.name = name;
    }

    public String getRequestLine() {
        return this.name + " has requested to do business with you";
    }

    public void handle() {
        char choice = readChoice();
        System.out.println(choice == 'y' ? "Approved" : "Rejected");
    }

    private char readChoice() {
        System.out.print("Approve request? (y/n): ");
        return In.nextChar();
    }
}

File: Tutor Demos/Week 7/Python/partnership_offer.py
from request import Request
class PartnershipOffer(Request):
    def __init__(self, name):
        self.__name = name
    
    def get_request_line(self):
        return f"{self.__name} has requested to do business with you"
    
    def handle(self):
        choice = input("Approve request? (y/n): ")
        print("Approved" if choice == "y" else "Rejected")